service/src/main/java/com/codeforcommunity/processor/authenticated/ProtectedSchoolProcessorImpl.java

```diff
@@ -734,6 +734,26 @@ public BookLog updateBookLog(
     return new BookLog(log.getId(), log.getCount(), log.getDate(), log.getNotes());
   }
 
+  @Override
+  public void deleteBookLog(JWTData userData, int schoolId, int bookId) {
+    if (!userData.isAdmin()) {
+      throw new AdminOnlyRouteException();
+    }
+    SchoolsRecord school = this.queryForSchool(schoolId);
+    if (school == null) {
+      throw new SchoolDoesNotExistException(schoolId);
+    }
+
+    BookLogsRecord log = db.selectFrom(BOOK_LOGS).where(BOOK_LOGS.ID.eq(bookId)).fetchOne();
```

Comment:
I would also recommend checking to see if the book has already been deleted when you query the DB query. There are examples of this in other routes